tests/performance/test_mongo_redis_performance.py

```diff
@@ -0,0 +1,197 @@
+
+
+def measure(func):
+    def wrapper(*args, **kwargs):
+        start = time.time()
+        values = func(*args, **kwargs)
+        end = time.time()
+        TestPerformance.time = end - start
+        print(f'Elapsed time: {TestPerformance.time}')
+        return values
+
+    return wrapper
+
+
+class TestPerformance:
+    time = 0
+
+    @pytest.fixture(autouse=True)
+    def _initialize(
+        self,
+        node_range,
+        word_range,
+        letter_range,
+        alphabet_range,
+        word_link_percentage,
+        letter_link_percentage,
+        seed,
+    ):
+        self.node_type = 'Concept'
+        self.node_range = [int(i) for i in node_range.split('-')]
+        self.word_range = [int(i) for i in word_range.split('-')]
+        self.letter_range = [int(i) for i in letter_range.split('-')]
+        self.alphabet_range = [int(i) for i in alphabet_range.split('-')]
+        self.word_link_percentage = word_link_percentage
+        self.letter_link_percentage = letter_link_percentage
+        self.seed = seed
+        if seed:
+            random.seed(seed)
+
+    @pytest.fixture(scope="class", autouse=True)
+    def database(self):
+        _db_up()
+        yield
+        _db_down()
+
+    @staticmethod
+    def combinatorial_loop(d: Dict[str, Any], f: Callable[[int, int], None], percentage: float):
+        for v in d.values():
+            v = list(v)
+            for i, _ in enumerate(v):
+                for j in range(i + 1, len(v)):
+                    if random.random() > percentage:
+                        continue
+                    f(v[i], v[j])
+
+    @measure
+    def generate_links_word(self, word_dict):
+        links_dict_word = {}
+        word_count = self.word_range[1] - self.word_range[0]
+        for v in word_dict.values():
+            v = list(v)
+            for i in range(0, len(v)):
+                for j in range(i + 1, len(v)):
+                    if random.random() > self.word_link_percentage:
+                        continue
+                    key = f'{min(v[i], v[j])}->{max(v[i], v[j])}'
+                    if key in links_dict_word:
+                        links_dict_word[key] += 1 / word_count
+                    else:
+                        links_dict_word[key] = 1 / word_count
+        return links_dict_word
+
+    @measure
+    def generate_nodes(self, das: DistributedAtomSpace = None):
+        node_list = []
+        node_names = {}
+        word_dict = {}
+        for N in range(*self.node_range):
+            node_list.append({'name': []})
+            word_list = []
+            for W in range(*self.word_range):
+                node_list[N]['name'].append([])
+                node_list[N]['name'][W - self.word_range[0]] = ''.join(
+                    [chr(97 + randint(*self.alphabet_range)) for L in range(*self.letter_range)]
+                )
+                word_list.append(node_list[N]['name'][W - self.word_range[0]])
```

Comment:
This statement is unecessarily complex. I believe you could have something like:

 ```
 w = ''.join([chr(97 + randint(*self.alphabet_range)) for L in range(*self.letter_range)]"
 ```
 
 and then use `w` in lines 92 and 95

Reply:
Fixed